Increase testing coverage on TextArea component

JoeButler7 opened the pull request:
<!--- Provide a general summary of the PR in the Title above -->

#### What does this PR do?
Increases testing coverage for `TextArea` by adding behavioral tests for event handlers. 
Coverage numbers before and after the tests were added are in the following table:
| Type       | Before | After  |
|------------|--------|--------|
| Statements | 73.81% | 95.24% |
| Branches   | 66.67% | 81.82% |
| Functions  | 71.43% | 92.86% |
| Lines      | 79.49% | 94.87% |
#### Where should the reviewer start?
`src/js/components/TextArea/__tests__/TextArea-test.js`
#### What testing has been done on this PR?
ran `yarn test TextArea-test` to ensure that coverage was increased and all the tests passed
#### How should this be manually tested?
See above
#### Screenshots (if appropriate)
Coverage before the tests were updated
<img width="1790" alt="Screen Shot 2020-07-01 at 5 14 58 PM" src="https://user-images.githubusercontent.com/49867133/86292835-6ff9a200-bbbf-11ea-8fb8-62187864b622.png">Coverage after
<img width="1786" alt="Screen Shot 2020-07-01 at 5 13 25 PM" src="https://user-images.githubusercontent.com/49867133/86292840-72f49280-bbbf-11ea-8894-a84e4ed031d8.png">

#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
No
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

jcfilben commented on src/js/components/TextArea/__tests__/TextArea-test.js:
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
+      const { container, getByPlaceholderText } = render(
+        <Grommet>
+          <TextArea
+            id="item"
+            name="item"
```
Do you need to set the id and name for this test?

jcfilben commented on src/js/components/TextArea/__tests__/TextArea-test.js:
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
+      );
+      fireEvent.focus(getByPlaceholderText('item'));
+      expect(container.firstChild).toMatchSnapshot();
+      expect(onFocus).toHaveBeenCalled();
```
```suggestion
      expect(onFocus).toHaveBeenCalledTimes(1);
```

jcfilben commented on src/js/components/TextArea/__tests__/TextArea-test.js:
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
+      const { container, getByPlaceholderText } = render(
+        <Grommet>
+          <TextArea
+            id="item"
+            name="item"
```
Same thing here with id and name, do they need to be set for the test

jcfilben commented on src/js/components/TextArea/__tests__/TextArea-test.js:
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
+        target: { value: 'o' },
+      });
+      expect(container.firstChild).toMatchSnapshot();
+      expect(onChange).toHaveBeenCalled();
```
```suggestion
      expect(onChange).toHaveBeenCalledTimes(1);
```

jcfilben commented on src/js/components/TextArea/__tests__/TextArea-test.js:
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
+      );
+      fireEvent.blur(getByPlaceholderText('item'));
+      expect(container.firstChild).toMatchSnapshot();
+      expect(onBlur).toHaveBeenCalled();
```
```suggestion
      expect(onBlur).toHaveBeenCalledTimes(1);
```

ShimiSun commented on src/js/components/TextArea/__tests__/TextArea-test.js:
```diff
@@ -163,5 +163,43 @@ describe('TextArea', () => {
+        </Grommet>,
+      );
+      fireEvent.change(getByPlaceholderText('item'), {
+        target: { value: 'o' },
```
The snapshot needs to be aligned with this new value.

JoeButler7 commented:
### Update:
changed name of tests and removed snapshots that were not showing any changes. In `onChange` removed snapshot and instead assert the value of the textArea after change event
